src/components/HomePage.js

```diff
@@ -122,7 +124,7 @@ export default React.createClass({
             return;
         }
 
-        if (!this.isAliasValid(entity) && !this.isUserIdValid(entity) && !this.isMsglinkValid(entity)) {
+        if (!this.isAliasValid(entity) && !this.isUserIdValid(entity) && !this.isMsglinkValid(entity) && !this.isRoomIdValid(entity)) {
```

Comment:
this is ugly - `getLinkType(entity)` which returns undef if there is no match would be preferable. but <shrug>